### PR TITLE
Normalize with engineered feats

### PR DESCRIPTION
**Pull request by TobyBoyne**
## Motivation

In #678, bounds were no longer provided to input transforms, and were instead learned from data. This was done to avoid explicitly providing the bounds of the engineered features.

However, this caused issues in #705, since I required an input to be strictly within the bounds [0, 1], but it was often scaled beyond that value if the training data didn't cover the full space. So instead, we revert back to the original behaviour for regular continuous features, and only learn the bounds for engineered features.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/experimental-design/bofire/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

I have adapted the existing test cases to verify the correct behaviour.

### CODE REVIEW

File: bofire/surrogates/utils.py [hunk 1]
```diff
@@ -97,23 +97,48 @@ def get_scaler(
...
 
     if scaler_type == ScalerEnum.NORMALIZE:
-        return Normalize(
-            d=d,
-            # bounds=torch.tensor([lower, upper]).to(**tkwargs),
-            indices=ord_dims,
-            batch_shape=torch.Size(),
+        # We create a separate Normalize for non-engineered features,
+        # since bounds are known for these features.
+        lower, upper = inputs.get_bounds(
```
Comment (jduerholt) [inline bofire/surrogates/utils.py:118]: This would introduce a bug, you have to use the actual training data here, because it can be that the training data is wider as the bounds. Have a look here: https://github.com/experimental-design/bofire/pull/678/changes#diff-0e88d06b923cb05ccccfbb3e27430a73e500ba21811ed5520ade8602b9a52bb7

where I changed the original behavior, have a look here.

File: bofire/surrogates/utils.py
Comment (jduerholt) [inline bofire/surrogates/utils.py:201]: Do not return a chained input transform but a dictionary of scalers that we then use here.

File: bofire/surrogates/utils.py [hunk 2]
```diff
@@ -61,7 +62,8 @@ def get_scaler(
     engineered_features: EngineeredFeatures,
     categorical_encodings: InputTransformSpecs,
     scaler_type: ScalerEnum,
-) -> Union[InputStandardize, Normalize, None]:
+    X: pd.DataFrame | None = None,
+) -> InputStandardize | Normalize | dict[str, Normalize] | None:
```
Comment (jduerholt) [inline bofire/surrogates/utils.py]: Can you always return a dictionary?
